fairscale/utils/parallel.py

```diff
@@ -15,6 +15,8 @@
 import torch.nn.functional as F
 
 
+# FIXME: this seems like a candidate for pytorch.
```

Comment:
see above. since we are separated from pytorch, we develop a bit differently from pytorch team's module.

* for pytorch team's module, they can assume a single version of pytorch, which is the "current" version they are running with.
* for us, we support multiple versions of pytorch (this make user's life a bit easier since they don't have to get new version of pytorch to get new features from us)
* on the other hand, we have a smaller code base and fewer tests, we can potentially develop faster since we don't have to checkin our code into a big codebase and run large number of tests just for our module.